File: trading/src/test/kotlin/io/squer/devenv/containers/PricesMock.java
package io.squer.devenv.containers;

import java.nio.file.Path;
import org.testcontainers.containers.GenericContainer;
import org.testcontainers.containers.Network;
import org.testcontainers.images.builder.ImageFromDockerfile;
import org.testcontainers.utility.DockerImageName;

public class PricesMock extends GenericContainer<PricesMock> {

  private static final String IMAGE_NAME = "squer-trading-mock";

  private static final int EXPOSED_PORT = 8090;

  private static final int HOST_PORT = 8090;

  private static final int CONTAINER_PORT = 8090;

  private PricesMock() {
    super(DockerImageName.parse(IMAGE_NAME));
  }

  private static GenericContainer<PricesMock> instance;

  public static GenericContainer<PricesMock> getInstance(final Network network) {
    if (instance == null) {
      instance = new PricesMock()
        .withEnv("KAFKA_BOOTSTRAP_SERVERS", "redpanda:29092")
        .withNetwork(network)
        .withNetworkMode("host")
        .withExposedPorts(EXPOSED_PORT)
        .withCreateContainerCmdModifier(cmd -> cmd.withHostName("prices-mock").withName("prices-mock"));

      ((PricesMock) instance).configurePorts();
    }
    return instance;
  }

  private void configurePorts() {
    super.addFixedExposedPort(HOST_PORT, CONTAINER_PORT);
  }

}

File: trading/src/test/kotlin/io/squer/devenv/containers/PricesMock.kt
package io.squer.devenv.containers

import com.github.dockerjava.api.command.CreateContainerCmd
import org.testcontainers.containers.GenericContainer
import org.testcontainers.containers.Network
import org.testcontainers.utility.DockerImageName

class PricesMock(network: Network?) : GenericContainer<PricesMock?>(DockerImageName.parse(IMAGE_NAME)) {
    init {
        addFixedExposedPort(HOST_PORT, CONTAINER_PORT)
        withEnv("KAFKA_BOOTSTRAP_SERVERS", "redpanda:29092")
        withNetwork(network)
        withNetworkMode("host")
        withCreateContainerCmdModifier { cmd: CreateContainerCmd -> cmd.withHostName("prices-mock") }
    }

    companion object {
        private const val IMAGE_NAME = "squer-trading-mock"
        private const val HOST_PORT = 8090
        private const val CONTAINER_PORT = 8090
    }
}
